### PR TITLE
UPSTREAM: <carry>: openshift: Allow to run disk delete ops async

### PR DESCRIPTION
**Pull request by ingvagabund**
**What this PR does / why we need it**:

So the machine controller can delete multiple disks in (pseudo-)parallel manner.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by ingvagabund** [comment]
Sharing actuator logs showing disk deletion is async now:

```
E0717 12:51:52.331736   13162 actuator.go:86] Machine error: failed to delete machine "worker-machineset-2rwls": failed to delete OS disk: result error: compute.DisksDeleteFuture: asynchronous operation has not completed
E0717 12:52:03.154484   13162 actuator.go:86] Machine error: failed to delete machine "master-machine": failed to delete OS disk: result error: compute.DisksDeleteFuture: asynchronous operation has not completed
```

TLDR;
```
I0717 12:50:58.655001   13162 actuator.go:120] Deleting machine master-machine
I0717 12:50:58.655014   13162 machine_scope.go:184] Decoding ProviderConfig from Value
I0717 12:50:58.656361   13162 virtualmachines.go:211] deleting vm master-machine 
E0717 12:51:00.085618   13162 actuator.go:86] Machine error: failed to delete machine "master-machine": failed to delete machine: compute.VirtualMachinesDeleteFuture: asynchronous operation has not completed
E0717 12:51:00.158022   13162 controller.go:208] Failed to delete machine "master-machine": requeue in: 1m0s
I0717 12:51:00.158047   13162 controller.go:352] Actuator returned requeue-after error: requeue in: 1m0s
I0717 12:51:00.158105   13162 controller.go:129] Reconciling Machine "worker-machineset-2rwls"
I0717 12:51:00.158117   13162 controller.go:298] Machine "worker-machineset-2rwls" in namespace "default" doesn't specify "cluster.k8s.io/cluster-name" label, assuming nil cluster
I0717 12:51:00.158128   13162 controller.go:193] Reconciling machine "worker-machineset-2rwls" triggers delete
I0717 12:51:00.158135   13162 actuator.go:120] Deleting machine worker-machineset-2rwls
I0717 12:51:00.158146   13162 machine_scope.go:184] Decoding ProviderConfig from Value
I0717 12:51:00.159660   13162 virtualmachines.go:211] deleting vm worker-machineset-2rwls 
E0717 12:51:01.762735   13162 actuator.go:86] Machine error: failed to delete machine "worker-machineset-2rwls": failed to delete machine: compute.VirtualMachinesDeleteFuture: asynchronous operation has not completed
E0717 12:51:01.826052   13162 controller.go:208] Failed to delete machine "worker-machineset-2rwls": requeue in: 1m0s
I0717 12:51:01.826083   13162 controller.go:352] Actuator returned requeue-after error: requeue in: 1m0s
I0717 12:51:50.305585   13162 controller.go:129] Reconciling Machine "worker-machineset-2rwls"
I0717 12:51:50.305705   13162 controller.go:298] Machine "worker-machineset-2rwls" in namespace "default" doesn't specify "cluster.k8s.io/cluster-name" label, assuming nil cluster
I0717 12:51:50.305768   13162 controller.go:193] Reconciling machine "worker-machineset-2rwls" triggers delete
I0717 12:51:50.305809   13162 actuator.go:120] Deleting machine worker-machineset-2rwls
I0717 12:51:50.305866   13162 machine_scope.go:184] Decoding ProviderConfig from Value
I0717 12:51:50.310265   13162 virtualmachines.go:211] deleting vm worker-machineset-2rwls 
I0717 12:51:51.453099   13162 virtualmachines.go:228] successfully deleted vm worker-machineset-2rwls 
I0717 12:51:51.453144   13162 disks.go:49] deleting disk worker-machineset-2rwls_OSDisk
E0717 12:51:52.331736   13162 actuator.go:86] Machine error: failed to delete machine "worker-machineset-2rwls": failed to delete OS disk: result error: compute.DisksDeleteFuture: asynchronous operation has not completed
E0717 12:51:52.357967   13162 controller.go:208] Failed to delete machine "worker-machineset-2rwls": requeue in: 1m0s
I0717 12:51:52.358028   13162 controller.go:352] Actuator returned requeue-after error: requeue in: 1m0s
I0717 12:52:00.158361   13162 controller.go:129] Reconciling Machine "master-machine"
I0717 12:52:00.158404   13162 controller.go:298] Machine "master-machine" in namespace "default" doesn't specify "cluster.k8s.io/cluster-name" label, assuming nil cluster
I0717 12:52:00.158427   13162 controller.go:193] Reconciling machine "master-machine" triggers delete
I0717 12:52:00.158444   13162 actuator.go:120] Deleting machine master-machine
I0717 12:52:00.158468   13162 machine_scope.go:184] Decoding ProviderConfig from Value
I0717 12:52:00.159700   13162 virtualmachines.go:211] deleting vm master-machine 
I0717 12:52:02.562290   13162 virtualmachines.go:228] successfully deleted vm master-machine 
I0717 12:52:02.562374   13162 disks.go:49] deleting disk master-machine_OSDisk
E0717 12:52:03.154484   13162 actuator.go:86] Machine error: failed to delete machine "master-machine": failed to delete OS disk: result error: compute.DisksDeleteFuture: asynchronous operation has not completed
E0717 12:52:03.230066   13162 controller.go:208] Failed to delete machine "master-machine": requeue in: 1m0s
I0717 12:52:03.230102   13162 controller.go:352] Actuator returned requeue-after error: requeue in: 1m0s
I0717 12:52:52.358232   13162 controller.go:129] Reconciling Machine "worker-machineset-2rwls"
I0717 12:52:52.358271   13162 controller.go:298] Machine "worker-machineset-2rwls" in namespace "default" doesn't specify "cluster.k8s.io/cluster-name" label, assuming nil cluster
I0717 12:52:52.358290   13162 controller.go:193] Reconciling machine "worker-machineset-2rwls" triggers delete
I0717 12:52:52.358301   13162 actuator.go:120] Deleting machine worker-machineset-2rwls
I0717 12:52:52.358320   13162 machine_scope.go:184] Decoding ProviderConfig from Value
I0717 12:52:52.359558   13162 virtualmachines.go:211] deleting vm worker-machineset-2rwls 
I0717 12:52:53.542630   13162 virtualmachines.go:228] successfully deleted vm worker-machineset-2rwls 
I0717 12:52:53.542651   13162 disks.go:49] deleting disk worker-machineset-2rwls_OSDisk
I0717 12:52:53.678049   13162 disks.go:65] successfully deleted disk worker-machineset-2rwls_OSDisk
I0717 12:52:53.678177   13162 networkinterfaces.go:178] deleting nic worker-machineset-2rwls-nic
I0717 12:52:54.525430   13162 networkinterfaces.go:197] successfully deleted nic worker-machineset-2rwls-nic
I0717 12:52:54.525467   13162 publicips.go:102] deleting public ip os4-common-worker-machineset-2rwls-publicip
I0717 12:53:05.237503   13162 publicips.go:121] deleted public ip os4-common-worker-machineset-2rwls-publicip
I0717 12:53:05.322144   13162 controller.go:227] Machine "worker-machineset-2rwls" deletion successful
I0717 12:53:05.322336   13162 controller.go:129] Reconciling Machine "master-machine"
I0717 12:53:05.322376   13162 controller.go:298] Machine "master-machine" in namespace "default" doesn't specify "cluster.k8s.io/cluster-name" label, assuming nil cluster
I0717 12:53:05.322399   13162 controller.go:193] Reconciling machine "master-machine" triggers delete
I0717 12:53:05.322415   13162 actuator.go:120] Deleting machine master-machine
I0717 12:53:05.322441   13162 machine_scope.go:184] Decoding ProviderConfig from Value
I0717 12:53:05.323789   13162 virtualmachines.go:211] deleting vm master-machine 
I0717 12:53:06.136941   13162 virtualmachines.go:228] successfully deleted vm master-machine 
I0717 12:53:06.136971   13162 disks.go:49] deleting disk master-machine_OSDisk
I0717 12:53:06.284711   13162 disks.go:65] successfully deleted disk master-machine_OSDisk
I0717 12:53:06.284820   13162 networkinterfaces.go:178] deleting nic master-machine-nic
I0717 12:53:07.147585   13162 networkinterfaces.go:197] successfully deleted nic master-machine-nic
I0717 12:53:07.147724   13162 publicips.go:102] deleting public ip os4-common-master-machine-publicip
I0717 12:53:17.840827   13162 publicips.go:121] deleted public ip os4-common-master-machine-publicip
I0717 12:53:18.003059   13162 controller.go:227] Machine "master-machine" deletion successful
```

**Comment by frobware** [comment]
I have tested this manually. Previously the autoscaler e2e tests were taking ~900s. Now I see: 

```console
• [SLOW TEST:505.151 seconds]
[Feature:Machines][Serial] Autoscaler should
/home/aim/go-projects/cluster-api-actuator-pkg/src/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/autoscaler/autoscaler.go:229
  scale up and down
  /home/aim/go-projects/cluster-api-actuator-pkg/src/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/autoscaler/autoscaler.go:230
------------------------------
SSSSSSSSSSSSS
Ran 1 of 16 Specs in 505.151 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 15 Skipped
--- PASS: TestE2E (505.15s)
```

**Comment by frobware** [comment]
/approve

https://github.com/openshift/cluster-api-provider-azure/pull/54#issuecomment-512262240
